### ISSUE TITLE
Add drop-down selectors with multiple values

### ISSUE THREAD

**Issue by enarroied** [Avaiga/taipy#1834]
### Description

Hi,
When I use selectors, there isn't a combination that lets me have a multiple-value selector that is also a drop-down. 

### Solution Proposed

Something like drop-down and check boxes for example. The following doesn't give an error but also doesn't "work":

```
tgb.selector
    (...
        multiple=True,
        dropdown=True,
        mode="check",)
```

**Another thing that could be great, is if the callback is only called after I select/unselect all my values (when I go out of the drop-down list). Right now, every time I check a box, it calls on_action, so if I check (or uncheck) a lot of boxes, it calls the Callback many times.**

### Impact of Solution

To give an example, I used a side pane to hold a long list of checkboxes, but it would have been quicker to have this drop-down list, and also more efficient. 

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by FlorianJacta** [comment]
Is your issue with checkboxes or dropdown+multiple in general? You should be able to create a selector with a dropdown, and multiple equal to True at the same time.

**Comment by enarroied** [comment]
Hi Florian,
I think this is what I mean: If your list of values is long (imagine, 50 values like the American states), and they're all selected, your dropdown becomes so crowded that it's almost "theoric". It would be great if you could have a selector that displays a label (eg "Select States"), and when you click on it, it drops down and you select the states. 

And even greater if it only calls the Callback when you're out of the selector (not each time you select/unselect a single value)
Checkboxes are not the most important, I like them, but it could be done with color gradients like the default. 

The feature request is more on the display than on the functional side. Except for the Callback behavior, that is functional.

I hope this makes sense :) Here is how a crowded list looks, it takes too much space in the app:
![drop-down](https://github.com/user-attachments/assets/78f9b991-de53-46e3-97d3-e6b7b26dc542)

**Comment by FlorianJacta** [comment]
I like it! I also had an issue with this. It would also be great to have a "All" option that will check everything

**Comment by KunjShah95** [comment]
Kidnly assign it to me

**Comment by KunjShah95** [comment]
Kindly assign it to me

**Comment by FlorianJacta** [comment]
[@KunjShah95](https://github.com/KunjShah95) You are already assigned to another issue. For hacktoberfest, we only assign issues one at a time. Please submit a PR on the other issue first, or remove your assignment.

Thank you.

**Comment by JagjeetChauhan** [comment]
@FlorianJacta can you assign this task to me.

**Comment by mohammed-anwar404** [comment]
hey @FlorianJacta can you assign this issue to me? 
I am a first-time contributor to Hactoberfest. I would want to work on this issue.

**Comment by FlorianJacta** [comment]
I will assign @JagjeetChauhan, as he was the first one to ask for this issue. Thank you for taking care of it

@mohammed-anwar404, there are many more issues on our repo!

**Comment by JagjeetChauhan** [comment]
Thankyou @FlorianJacta

**Comment by durveshwagh11** [comment]
pls ssign it to me @FlorianJacta

**Comment by FlorianJacta** [comment]
@durveshwagh11, sorry, this issue has already been assigned. You have many more to take in this repo

**Comment by sajithcode** [comment]
assign it to me

**Comment by ruhi47** [comment]
Hey, If the assigned person is inactive, I can take up the issue.
Thank you!

**Comment by FlorianJacta** [comment]
You can also work on it @ruhi47

**Comment by akanshSirohi** [comment]
@FlorianJacta If inactive, I can resolve this issue.

**Comment by FlorianJacta** [comment]
Here you go @akanshSirohi

**Comment by Manya1103** [comment]
please assign it to me

**Comment by akanshSirohi** [comment]
@FlorianJacta Any suggestions on this approach? I'm just getting started and exploring the project, so I came across this approach we can use.
<table>
    <tr><td>
      <img src="https://github.com/user-attachments/assets/5de375f0-9913-4c65-845a-d5659914bd94" />
    </td><td>
      <img src="https://github.com/user-attachments/assets/24f7f772-f408-4923-9e26-d67c2574dd84" />
    </td></tr>
</table>

**Comment by FlorianJacta** [comment]
That could do the trick! What do you think @FabienLelaquais ?

**Comment by akanshSirohi** [comment]
@FlorianJacta, you can check out the PR [https://github.com/Avaiga/taipy/pull/2168](https://github.com/Avaiga/taipy/pull/2168).

I have added functionality to support the requested feature. Now, this syntax can be used:
```
tgb.selector(
    ...
    multiple=True,
    dropdown=True,
    mode="check",
)
```
Any suggestions would be helpful for me to continue working on this, including anything else I might need to add or change.

**Comment by FlorianJacta** [comment]
Can you create the PR and mention @FredLL-Avaiga and @FabienLelaquais ?

**Comment by akanshSirohi** [comment]
Sure

**Comment by FredLL-Avaiga** [comment]
Would it work if we define a new property that would make the dropdown selection display only the number of selected items
`show_multiple_selected: boolean`
`True`: actual way of displaying selection
`False`: display the number of selected items

@enarroied @FlorianJacta @FabienLelaquais ?

**Comment by FabienLelaquais** [comment]
Had a short talk with @FredLL-Avaiga.
@enarroied, @FlorianJacta please give your feedback on what we propose.

- Regarding the crowded field when many items are selected.
  We propose to provide a new dynamic property: a string provided by the user, which would be displayed instead of the list of chips should a given condition be met.
  I.E. : you want to display a message if more than 5 items are selected.
    In the `on_change` callback, the user would set this property like this:
```
       if var_name == "my_selection":
           state.my_selector_message = "{len(var_value)} states selected" if len(var_value) > 5 else None
```
  The property name has still to be defined. `message` seems a bit vague to me.

- Regarding the postponing of the callback invocation.
  We propose to provide a new property `apply_button` (?)  that if set to True (and the selector is dropdown and muliple), will add a 'Apply' button to the UI. The action of this button will close the popup and invoke the `on_change` callback.

- We also have to decide anything on a 'Select all'/'Select None' button of some sort that to me makes a lot of sense.

What do you think?

**Comment by FlorianJacta** [comment]
This seems great to me. Would we lose the styling when just a few values are selected? Normally, values are put inside a little container.

I would love to have this "Select All"/"Select None" button; this happens a lot in apps

**Comment by enarroied** [comment]
HI. Yes I like the ideas. 
Select All / None would be a nice improvement.
Will users be able to exit the drop-down selector after changing selections but without validating them (and then no callback is triggered either)?

**Comment by FabienLelaquais** [comment]
> This seems great to me. Would we lose the styling when just a few values are selected? Normally, values are put inside a little container.
> 

The legacy behavior needs to be maintained. So I guess the answer is yes.
I mean no. 
Well you get the point right?

> I would love to have this "Select All"/"Select None" button; this happens a lot in apps
Sure. That's a bit tricky in terms of UI though. but ok.

**Comment by FabienLelaquais** [comment]
> HI. Yes I like the ideas. Select All / None would be a nice improvement. Will users be able to exit the drop-down selector after changing selections but without validating them (and then no callback is triggered either)?

I guess click outside / ESC can do that.
But then... your variable is not aligned with the selection.
This action should not impact the selection bound variable whatsoever... and your selection is lost (remains what it was when you first opened the popup).
Would that work for you?

**Comment by enarroied** [comment]
Yes, I think that if you don't trigger the callback, the bound variables shouldn't be updated; both go together. The only "risk" I see with losing selection clicking outside is that it could be less intuitive, but I think users would get used to it. 
You could also have the opposite, a "Cancel" button (or another name) that leaves the selection as it was before and doesn't trigger the callback. And let the variables update and the callback trigger happen when the user closes the selector. But either way makes sense to me.

Also, I think the the select all / none should be pure front-end elements, that check/uncheck all boxes. The reason is that we often use these to clear a selection to select a new set, or we select all to then select "all but one".

**Comment by FredLL-Avaiga** [comment]
I'm not at ease with this selected but not consistent state ...

**Comment by JonSpeare** [comment]
Just wanted to give my input.
I know this is a bit of a big ask, but having used multiple applications for the data and table with service stations, I have to say that Airtable's Multiselect is by far the best I have found as shown in this picture.

![Image](https://github.com/user-attachments/assets/8df17e7f-c5a9-48a0-9b18-0d52a6614707)

It’s probably the last feature that’s preventing me from purging Airtable from our business because I know our CFO would go nuts if he didn’t have this filter.

No overflow is a good compromise for large selctions like [enarroied](https://github.com/enarroied) pointed out.
Searchable list is a must large amount of options.
The Easy to understand checkbox for what is elected is generaly easy to avoid confusion.
the Clear all is usefull more often than one would think.
And the spesification / enhanced filter option at button is just awsome.

I found that every user of our tables just "gets" how to use it.
